### PR TITLE
agentless: migrate server-acl-init to use server connection manager

### PR DESCRIPTION
**Pull request by thisisnotashwin**
Changes proposed in this PR:

- Integrate server-acl-init with consul-server-connection-manager to enable it to work in agentless environments
- Migrate usages of os.Remove() to os.RemoveAll()

How I've tested this PR:
acceptance and unit tests

How I expect reviewers to test this PR:
👀

Checklist:
- [ ] Tests added
- [ ] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: charts/consul/templates/_helpers.tpl [hunk 1]
```diff
@@ -353,6 +353,9 @@ Consul server environment variables for consul-k8s commands.
 {{- if and .Values.externalServers.enabled .Values.externalServers.tlsServerName }}
 - name: CONSUL_TLS_SERVER_NAME
   value: {{ .Values.externalServers.tlsServerName }}
+{{- else if .Values.global.cloud.enabled }}
+- name: CONSUL_TLS_SERVER_NAME
+  value: server.{{ .Values.global.datacenter}}.{{ .Values.global.domain}}
```
Comment (ishustava) [inline charts/consul/templates/_helpers.tpl:358]: I don't think we need to pass this one anymore. server-conn-mgr should set TLS server name if the address is a DNS name.

File: control-plane/consul/consul.go [hunk 2]
```diff
@@ -10,7 +10,7 @@ import (
 	capi "github.com/hashicorp/consul/api"
 )
 
-//go:generate mockery --name ServerConnectionManager --inpackage
+//go:generate mockery --name ServerConnectionManager --inpkg
```
Comment (ishustava) [inline control-plane/consul/consul.go:13]: nice!

File: control-plane/subcommand/server-acl-init/command.go [hunk 3]
```diff
@@ -348,30 +335,44 @@ func (c *Command) Run(args []string) int {
...
+	var watcher consul.ServerConnectionManager
+	serverConnMgrCfg, err := c.consulFlags.ConsulServerConnMgrConfig()
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("unable to create config for consul-server-connection-manager: %s", err))
+		return 1
 	}
+	serverConnMgrCfg.Credentials.Type = discovery.CredentialsTypeStatic
+	serverConnMgrCfg.Credentials.Static = discovery.StaticTokenCredential{Token: bootstrapToken}
+	if c.watcher == nil {
+		watcher, err = discovery.NewWatcher(c.ctx, serverConnMgrCfg, c.log)
```
Comment (ishustava) [inline control-plane/subcommand/server-acl-init/command.go:355]: Could we name this logger 🙏 ?

**Review by ishustava — APPROVED** [review]
Looks great ❤️  Just a few minor comments

**Review by curtbushko — APPROVED** [review]
Fantastic!

**Pull request merged**